backend/interactive/mixplay.js

```diff
@@ -391,6 +396,23 @@ mixplayClient.state.on('participantJoin', async participant => {
     }
 });
 
+async function getCurrentUsers() {
+    let participantMap = mixplayClient.state.getParticipants();
+    let participants = Array.from(participantMap);
+    let participantNames = [];
+
+    for (let user in participants) {
+        if (user != null) {
+            user = participants[user][1];
+            if (user.anonymous !== true) {
+                participantNames.push(user);
+            }
+        }
+    }
+
+    return participantNames;
+}
```

Comment:
You could shorten this method to:
```js
function getConnectedUsernames() {
    let participants = [...mixplayClient.state.getParticipants().values()];
    return participants
        .filter(p => p != null && !p.anonymous)
        .map(p => p.username);
}
```
Note I also renamed the method to make it more clear that this only returns usernames and not entire user/participant objects when calling it elsewhere in the software.
Lastly, it does not need to be `async` as nothing in this method needs to `await` on a Promise.

Reply:
Oh wow, so that participants variable turns the Map into an array? I'll have to research the [...] thing.

Reply:
It’s called the Spread operator. You can read more about it here: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Spread_syntax 
It’s super useful. 

And to be clear, it’s spreading `getParticipants().values()`. So it’s not spreading the Map but what `.values()` returns (an Iterable object containing the values of the map). 
Ultimately `let valuesArray = [...map.values()];` is the same as `let valuesArray = Array.from(map.values())`, it’s just the newer (and nicer imo) way.